sql/src/main/java/org/apache/druid/sql/calcite/aggregation/builtin/EarliestLatestAnySqlAggregator.java

```diff
@@ -206,19 +195,74 @@ public Aggregation toDruidAggregation(
       );
     }
 
+    final String fieldName = getColumnName(plannerContext, virtualColumnRegistry, args.get(0), rexNodes.get(0));
+
+    final AggregatorFactory theAggFactory;
+    switch (args.size()) {
+      case 1:
+        theAggFactory = aggregatorType.createAggregatorFactory(aggregatorName, fieldName, null, outputType, -1);
+        break;
+      case 2:
+        if (outputType.anyOf(ValueType.STRING, ValueType.COMPLEX)) {
```

Comment:
nit: might be worth a comment that this is to handle 2nd argument as either byte size or time column, depending on the input type to make it a bit easier to make sense of what's going on here.

I think you could also potentially just make this check be `!outputType.isNumeric()` since those are fixed width and don't have a size limit

Reply:
the if check you mention seems better to me - changed to that. also, inlined the signature in the if-else block for better readability